pkg/extension/extension.go

```diff
@@ -0,0 +1,22 @@
+package extension
+
+import (
+	"context"
+
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type Extension interface{}
+
+type PreReconciliationExtension interface {
+	PreReconcile(ctx context.Context, obj *unstructured.Unstructured, release release.Release, vals chartutil.Values) error
+}
+type PreDeletionExtension interface {
+	PreDelete(ctx context.Context, obj *unstructured.Unstructured, release release.Release, vals chartutil.Values) error
+}
+
```

Comment:
No need for a PostDelete extension? 🤔

Reply:
Not sure. I believe *we* don't have this use-case currently. Do you just have a feeling that it is missing or do you have a strong example? Besides, we can also ask Joe for his input on this.

But, the nice thing about this approach is that we can easily add it without breaking anything.

Reply:
ok